src/PersonObjects/Sleeve/Sleeve.ts

```diff
@@ -859,10 +859,8 @@ export class Sleeve extends Person {
    * Returns boolean indicating success
    */
   workForFaction(p: IPlayer, factionName: string, workType: string): boolean {
-    if (factionName === "") {
-      return false;
-    }
-    if (!(Factions[factionName] instanceof Faction) || !p.factions.includes(factionName)) {
+    const faction = Factions[factionName]
+    if (factionName === "" || !faction || !(faction instanceof Faction) || !p.factions.includes(factionName)) {
```

Comment:
Doesn't `!p.factions.includes(factionName)` cover the `factionName === ""` case? Or is it possible for the former to return `false` when the latter would return `true`? Same goes for `!faction` and `!(faction instanceof Faction)`

Reply:
yeah i had these thoughts too... but then i'm not sure how some users could end up running into the issue further down (the .getInfo on undefined).

 maybe by storing the faction earlier now it wouldn't be an issue.

i guess i was thinking well, it might be redundant but better safe than sorry.

issue is this bug is a PIA to replicate even with user save files it seems.